tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -51,75 +36,98 @@
-        UIButton* cellButton = [cell.subviews objectAtIndex:2];
-        LOG_INFO(@"Cell subview[1] text: %@", cellLabel.text);
-
-        // We have to artificially ref the element since the block needs to keep it around
-        backingElement = [cellButton xamlElement];
-        [backingElement retain];
-
-        // Register callback and wait for the property changed event to trigger
-        int64_t callbackToken = [backingElement
-            registerPropertyChangedCallback:[WXCControl backgroundProperty]
-                            callback:^(WXDependencyObject* sender, WXDependencyProperty* dp) {
+    dispatch_sync(dispatch_get_main_queue(), ^{
+        // Extract UIButton.titleLabel control to verify its visual state
+        WXFrameworkElement* titleElement = [buttonVC.button.titleLabel xamlElement];
+        ASSERT_TRUE(titleElement);
+
+        // Register RAII event subscription handler
+        XamlEventSubscription
+            xamlSubscriber(titleElement, [WXCTextBlock foregroundProperty], ^(WXDependencyObject* sender, WXDependencyProperty* dp) {
+                // Validate the state of the XAML object
                 WUXMSolidColorBrush* solidBrush = rt_dynamic_cast([WUXMSolidColorBrush class], [sender getValue:dp]);
-                ASSERT_TRUE(solidBrush);
+                LOG_INFO("XAML element color (rgba): %d,%d,%d,%d",
+                         [solidBrush.color r],
+                         [solidBrush.color g],
+                         [solidBrush.color b],
+                         [solidBrush.color a]);
 
-                LOG_INFO("Backing XAML element backgroundColor (rgba): %d,%d,%d,%d",
-                    [solidBrush.color r],
-                    [solidBrush.color g],
-                    [solidBrush.color b],
-                    [solidBrush.color a]);
+                // Extract the title color for the normal state
+                UIColor* titleColorNormal = [buttonVC.button titleColorForState:UIControlStateNormal];
 
                 CGFloat red, green, blue, alpha;
-                [cellButton.backgroundColor getRed:&red green:&green blue:&blue alpha:&alpha];
-                LOG_INFO("UIButton.backgroundColor (rgba): %.2f,%.2f,%.2f,%.2f", red, green, blue, alpha);
+                [titleColorNormal getRed:&red green:&green blue:&blue alpha:&alpha];
+                LOG_INFO("UIButton.titleColorForState:normal (rgba): %.2f,%.2f,%.2f,%.2f", red, green, blue, alpha);
 
                 // Validate that the change is reflected on the backing XAML control
                 EXPECT_EQ_MSG(solidBrush.color.r, (int)(red * 255), @"Failed to match red component");
                 EXPECT_EQ_MSG(solidBrush.color.g, (int)(green * 255), @"Failed to match green component");
                 EXPECT_EQ_MSG(solidBrush.color.b, (int)(blue * 255), @"Failed to match blue component");
                 EXPECT_EQ_MSG(solidBrush.color.a, (int)(alpha * 255), @"Failed to match alpha component");
```

Comment:
Yes, that is correct. I took this shortcut because I didn't implement the discrete combobox yet. I could move to a UITextfield and pass through blueColor, redColor etc but I also wanted to leverage another common control such as UISlider for our pattern.

---
In reply to: [97414580](https://github.com/Microsoft/WinObjC/pull/1771#discussion_r97414580) [](ancestors = 97414580,97414376)